week4/filter-less/helpers.c

```diff
@@ -0,0 +1,115 @@
+            int sepiaGreen = round(.349 * originalRed + .686 * originalGreen + .168 * originalBlue);
+            int sepiaRed = round(.393 * originalRed + .769 * originalGreen + .189 * originalBlue);
+
+            bool blueHasOverflow = sepiaBlue > 255;
+            bool greenHasOverflow = sepiaGreen > 255;
+            bool redHasOverflow = sepiaRed > 255;
+
+            if (blueHasOverflow) {
+                sepiaBlue = 255;
+            }
+
+            if (greenHasOverflow) {
+                sepiaGreen = 255;
+            }
+
+            if (redHasOverflow) {
+                sepiaRed = 255;
+            }
+
+            image[sepiaHeight][sepiaWidth].rgbtBlue = sepiaBlue;
+            image[sepiaHeight][sepiaWidth].rgbtGreen = sepiaGreen;
+            image[sepiaHeight][sepiaWidth].rgbtRed = sepiaRed;
+        }
+    }
+
+    return;
+}
+
+void reflect(int height, int width, RGBTRIPLE image[height][width]) {
+    for (int reflectedHeight = 0; reflectedHeight < height; reflectedHeight++) {
+        for (int reflectedWidth = 0; reflectedWidth < width; reflectedWidth++) {
+            image[reflectedHeight][reflectedWidth] = image[reflectedHeight][width * 2 - reflectedWidth];
+        }
+    }
+
+    return;
+}
+
+void blur(int height, int width, RGBTRIPLE image[height][width]) {
+    int averageBlue;
+    int averageGreen;
+    int averageRed;
+
+    for (int blurHeight = 0; blurHeight < height; blurHeight++) {
+        for (int blurWidth = 0; blurWidth < width; blurWidth++) {
+            int validCounter = 1;
+            int originalBlueValue = image[blurHeight][blurWidth].rgbtBlue;
+            int originalGreenValue = image[blurHeight][blurWidth].rgbtGreen;
+            int originalRedValue = image[blurHeight][blurWidth].rgbtRed;
+            int currentBlurBlueValue;
+            int currentBlurGreenValue;
+            int currentBlurRedValue;
+            int sumOfBlue = originalBlueValue;
+            int sumOfGreen = originalGreenValue;
+            int sumOfRed = originalRedValue;
+            
+            for (int heightIterator = -1; heightIterator <= 1; heightIterator++) {
+                for (int widthIterator = -1; widthIterator <= 1; widthIterator++) {
+                    bool adjacentPixelIsValid = (blurHeight + heightIterator) >= 0 && (blurHeight + heightIterator) < height && (blurWidth + widthIterator) >= 0 && (blurWidth + widthIterator) < width;
+                    bool isActuallyOriginalValue = heightIterator == blurHeight && widthIterator == blurWidth;
+
+                    if (adjacentPixelIsValid) {
+                        currentBlurBlueValue = image[blurHeight + heightIterator][blurWidth + widthIterator].rgbtBlue;
+                        currentBlurGreenValue = image[blurHeight + heightIterator][blurWidth + widthIterator].rgbtGreen;
+                        currentBlurRedValue = image[blurHeight + heightIterator][blurWidth + widthIterator].rgbtRed;
+                    }
```

Comment:
```suggestion
                    bool adjacentPixelIsValid = (blurHeight + heightIterator) >= 0 && (blurHeight + heightIterator) < height && (blurWidth + widthIterator) >= 0 && (blurWidth + widthIterator) < width;
                    if (!adjacentPixelIsValid) {
                        continue;
                    }
                    
                    bool isActuallyOriginalValue = heightIterator == blurHeight && widthIterator == blurWidth;

                    currentBlurBlueValue = image[blurHeight + heightIterator][blurWidth + widthIterator].rgbtBlue;
                    currentBlurGreenValue = image[blurHeight + heightIterator][blurWidth + widthIterator].rgbtGreen;
                    currentBlurRedValue = image[blurHeight + heightIterator][blurWidth + widthIterator].rgbtRed;
```
An early exit is more readable and performant for these kind of conditions.